spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/registry/UriRegistryPopulator.java

```diff
@@ -73,6 +72,15 @@ public void setResourceLoader(ResourceLoader resourceLoader) {
 				for (String key : properties.stringPropertyNames()) {
 					try {
 						URI uri = new URI(properties.getProperty(key));
+						if (!overwrite) {
+							try  {
+								registry.find(key);
+							}
+							catch (IllegalArgumentException e) {
+								// this key already exists
```

Comment:
@markfisher Pushed some changes to my branch [here] (https://github.com/ilayaperumalg/spring-cloud-deployer/commit/2ea3d30d8d3b546c6ab250f97baac0f9766ee015) for the above comment with a test that showcases it. If that looks good to you, I can merge this PR along with those changes.